drivers/gpio/gpio_ambiq.c

```diff
@@ -70,18 +105,116 @@ static int ambiq_gpio_pin_configure(const struct device *dev, gpio_pin_t pin, gp
 		pincfg.GP.cfg_b.eCEpol = AM_HAL_GPIO_PIN_CEPOL_ACTIVELOW;
 		am_hal_gpio_state_write(pin, AM_HAL_GPIO_OUTPUT_CLEAR);
 	}
-
+#endif
 	am_hal_gpio_pinconfig(pin, pincfg);
 
 	return 0;
 }
 
+#if defined(CONFIG_SOC_SERIES_APOLLO3X)
+uint32_t am_hal_gpio_pinconfig_get(uint32_t ui32GpioNum, am_hal_gpio_pincfg_t *psGpioCfg)
+{
+	uint32_t ui32GPCfgAddr, ui32PadregAddr, ui32AltpadAddr;
```

Comment:
Please use Zephyr naming style.
https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style
https://kernel.org/doc/html/latest/process/coding-style.html

Reply:
Actually, the function should be put into our hal. Since it is a new addition, it would't cause any disturbance of the old one. Then we could avoid changing any code all together. Plus error codes are all our HAL return codes, all the more reason to be put into HAL.

Reply:
Agreed, I will open a ticket internally if you haven't already.

Reply:
> Edit: if this is for apollo3, I will let Hao handle the HAL update. Do we think this is also necessary for the Apollo4/5?

No, 4/5 already have it.

Reply:
@RichardSWheatley apollo4/5 already have am_hal_gpio_pinconfig_get API in HAL since their different register designs.

Reply:
my original idea is to keep those changes in zephyr driver and not modify apollo3 HAL now, considering apollo3 is pretty old generation and there is no plan to have sdk release for it in the near future, changing the HAL here will make it different from our ambiqsuite sdk. To be note is that the ambiq_hal in zephyr comes from last year's v3.1.1 sdk, ideally we should only update HAL in zephyr once new ambiqsuite sdk released and upgrade the entire HAL, it's hard to maintainance changes in two different repos.
anyway, if you all insist to put the changes to HAL now, it's also ok, but I need to create PR to ambiqsuite first to be consistence.

Reply:
I think it is OK maintainance wise because if you forgot to add back the new stuff, we would know immediately as they are used in Zephyr and would stop compile.